### PR TITLE
MESH-1257: Runtime upgrade functionality for CDD_ID.

### PR DESCRIPTION
**Pull request by Centril**
Review notes: Take a look at the new doc comment for `#[derive(Migrate)]` first so you understand how things are intended to work before reviewing the code.

### CODE REVIEW

**Comment by CJP10** [comment]
LGTM!

**Review by CJP10 — APPROVED** [review]

File: primitives_derive/src/lib.rs [hunk 1]
```diff
@@ -76,7 +78,33 @@ pub fn deserialize_u8_strong_typed_derive(input: TokenStream) -> TokenStream {
...
+/// #[derive(..., Encode, Migrate)]
+/// struct Foo {
+///     #[migrate_with(bar + 1)]
+///     bar: u8,
+/// }
+/// ```
+///
+/// By default, `#[derive(Migrate)]` will use `Empty` for `type Context`.
+/// This can however be changed by specifying the type with `#[migrate_context(TheContextType)]`.
+#[proc_macro_derive(
```
Comment (vkomenda) [inline primitives_derive/src/lib.rs:104]: Can you add docs for `migrate_context`? It's not apparent why `#[derive(Migrate)]` benefits from it.

**Comment by adamdossa** [comment]
FYI - I tried updating an alcyone network using this branch, and it seems to work correctly. We will need to do more testing before upgrading the public testnet, but for now this looks good to me.

**Review by adamdossa — APPROVED** [review]

File: primitives_derive/src/lib.rs (hunk 1 above)
Comment (Centril) [reply primitives_derive/src/lib.rs:107]: Added some more docs. :)

**Review by maxsam4 — APPROVED** [review]

**Pull request merged**